exonum-java-binding-core/src/main/java/com/exonum/binding/blockchain/Block.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.blockchain;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.google.auto.value.AutoValue;
+
+/**
+ * Exonum block header data structure.
+ *
+ * <p>A block is essentially a list of transactions, which is a result of the consensus algorithm
+ * (thus authenticated by the supermajority of validators) and is applied atomically to the
+ * blockchain state.
+ *
+ * <p>This structure only contains the amount of transactions and the transactions root hash as well
+ * as other information, but not the transactions themselves.
+ */
+@AutoValue
+public abstract class Block {
+
+  public static Block valueOf(
```

Comment:
7 parameters of only two types (3 integers and 4 hashcodes) beg for a builder (may be done separately, but add a TODO with a jira ref)

Reply:
Will do!